### PR TITLE
Allow customizing of Switch thumbs

### LINKED ISSUE #2621: No way to customize switch thumb painter
One of our customers would like to draw icons inside the thumb of the switch to indicate its state.

Not strictly material, but seems like a reasonable request and not something they should have to fork the entire switch.dart for.

Looking briefly today looks like we could pass a _thumbPainter to Switch and pass it down.

### PR DESCRIPTION
**Pull request by abarth**
Fixes #2621

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/material/switch.dart [hunk 1]
```diff
@@ -202,12 +247,15 @@ class _RenderSwitch extends RenderToggleable {
 
     paintRadialReaction(canvas, thumbOffset);
 
+    Color thumbColor = isActive ? Color.lerp(inactiveColor, activeColor, position.value) : inactiveColor;
     if (_cachedThumbColor != thumbColor) {
```
Comment (Hixie) [inline packages/flutter/lib/src/material/switch.dart]: what if the thumb decorations changed but not the colour?

File: packages/flutter/lib/src/material/switch.dart [hunk 2]
```diff
@@ -202,12 +247,15 @@ class _RenderSwitch extends RenderToggleable {
...
-      _thumbPainter = new BoxDecoration(
-        backgroundColor: thumbColor,
-        shape: BoxShape.circle,
-        boxShadow: elevationToShadow[1]
-      ).createBoxPainter();
+      if (_inactiveThumbDecoration == null && _activeThumbDecoration == null) {
+        _thumbPainter = _createDefaultThumbDecoration(thumbColor).createBoxPainter();
+      } else {
+        Decoration startDecoration = _inactiveThumbDecoration ?? _createDefaultThumbDecoration(inactiveColor);
+        Decoration endDecoration = _activeThumbDecoration ?? _createDefaultThumbDecoration(isActive ? activeTrackColor : inactiveColor);
```
Comment (Hixie) [inline packages/flutter/lib/src/material/switch.dart]: activeColor, not activeTrackColor, no?

**Comment by Hixie** [comment]
travis is angry

**Comment by abarth** [comment]
I'm going to try this a different way.

**Pull request closed without merging**